watertap/property_models/tests/test_multicomp_aq_sol_prop_pack.py

```diff
@@ -1341,3 +1337,89 @@ def test_elec_properties_errormsg(model6):
         match="""Missing a valid trans_num_data configuration to build "trans_num_phase_comp" """,
     ):
         m[2].fs.stream[0].trans_num_phase_comp
+
+
+@pytest.fixture(scope="module")
+def model7():
+    m_hl = ConcreteModel()
+
+    m_hl.fs = FlowsheetBlock(dynamic=False)
+
+    m_hl.fs.properties = MCASParameterBlock(
+        solute_list=["A", "B", "C", "D", "E", "F"],
+        charge={"E": 1, "F": -1},
+        diffus_calculation=DiffusivityCalculation.HaydukLaudie,
+        molar_volume_data={
+            ("Liq", "A"): 96e-6,  # tested for benzene
+            ("Liq", "B"): 100e-6,  # arbitrary
+            ("Liq", "C"): 60e-6,  # arbitrary
+            ("Liq", "D"): 200e-6,  # arbitrary
+        },
+        diffusivity_data={("Liq", "E"): 1.33e-9, ("Liq", "F"): 2.03e-9},
+    )
+    # build state block
+    m_hl.fs.sb = m_hl.fs.properties.build_state_block([0], defined_state=True)
+    # touch on demand var when hayduklaudie is selected
+    m_hl.fs.sb[0].diffus_phase_comp
+    m_hl.fs.properties.visc_d_phase["Liq"] = 1.0e-3
+
+    return m_hl
+
+
+@pytest.mark.unit
+def test_diffus_hl(model7):
+    m = model7
+
+    assert (
+        m.fs.properties.config.diffus_calculation == DiffusivityCalculation.HaydukLaudie
+    )
+
+    check_dof(m, fail_flag=True)
+    assert_units_consistent(m)
+
+    # init and solve
+    m.fs.sb.initialize()
+    calculate_scaling_factors(m.fs)
+    results = solver.solve(m)
+    assert_optimal_termination(results)
+
+    assert isinstance(m.fs.properties.molar_volume_comp, Param)
+    assert m.fs.properties.molar_volume_comp["Liq", "A"].value == 96e-6
+    assert m.fs.properties.molar_volume_comp["Liq", "B"].value == 100e-6
+    assert m.fs.properties.molar_volume_comp["Liq", "C"].value == 60e-6
+    assert m.fs.properties.molar_volume_comp["Liq", "D"].value == 200e-6
+
+    sb = m.fs.sb[0]
+
+    assert sb.diffus_phase_comp["Liq", "A"].value == pytest.approx(
+        9.015e-10, rel=1e-3
+    )  # tested for benzene
+    assert sb.diffus_phase_comp["Liq", "B"].value == pytest.approx(8.801e-10, rel=1e-3)
+    assert sb.diffus_phase_comp["Liq", "C"].value == pytest.approx(1.189e-09, rel=1e-3)
+    assert sb.diffus_phase_comp["Liq", "D"].value == pytest.approx(5.851e-10, rel=1e-3)
+    assert sb.diffus_phase_comp["Liq", "E"].value == pytest.approx(1.33e-09, rel=1e-3)
+    assert sb.diffus_phase_comp["Liq", "F"].value == pytest.approx(2.03e-09, rel=1e-3)
+
+
+@pytest.mark.unit
+def test_diffus_properties_errormsg():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    with pytest.raises(
+        ConfigurationError,
+        match=(
+            "Insufficient data is provided to determine diffusivity of all solute components. "
+            "Check the molar_volume_data and diffusivity_data configuration."
```

Comment:
I think it would be beneficial for this to be more descriptive. For example, the user needs to add diffusivity data. Can you make this more digestible for the novice user he may not know what this means exactly without digging?

Reply:
Also should diffusivity data only be supplied for ions?